modules/storage/mysql-single-server/main.bicep

```diff
@@ -0,0 +1,331 @@
+
+@description('Optional. Status showing whether the server enabled infrastructure encryption..')
+@allowed(['Enabled', 'Disabled'])
+param infrastructureEncryption string = 'Disabled'
+
+@description('Optional. List of firewall rules to create on server.')
+param firewallRules firewallRulesType[] = []
+
+@description('Optional. List of virtualNetworkRules to create on mysql server.')
+param virtualNetworkRules virtualNetworkRuleType[] = []
+
+@description('Optional. List of privateEndpoints to create on mysql server.')
+param privateEndpoints array = []
+
+@description('Optional. Enable or disable geo-redundant backups. It requires at least a GeneralPurpose or MemoryOptimized skuTier.')
+@allowed(['Enabled','Disabled'])
+param geoRedundantBackup string = 'Disabled'
+
+@description('Optional. Enforce a minimal Tls version for the server.')
+@allowed(['TLS1_0', 'TLS1_1', 'TLS1_2', 'TLSEnforcementDisabled'])
+param minimalTlsVersion string = 'TLS1_2'
+
+var sslEnforcement = (minimalTlsVersion == 'TLSEnforcementDisabled') ? 'Disabled' : 'Enabled'
+
+@description('Optional. Restore point creation time (ISO8601 format), specifying the time to restore from.')
+param restorePointInTime string = ''
+
+@description('Optional. Whether or not public network access is allowed for this server.')
+@allowed(['Enabled','Disabled'])
+param publicNetworkAccess string = 'Disabled'
+
+@description('Optional. The name of the sku, typically, tier + family + cores, e.g. B_Gen4_1, GP_Gen5_8.')
+param skuName string = 'GP_Gen5_2'
+
+@description('Optional. Azure database for MySQL compute capacity in vCores (2,4,8,16,32)')
+param skuCapacity int = 2
+
+@description('Optional. Azure database for MySQL Sku Size ')
+param SkuSizeMB int = 5120
+
+@description('Optional. Azure database for MySQL pricing tier')
+@allowed([
+  'Basic'
+  'GeneralPurpose'
+  'MemoryOptimized'
+])
+param SkuTier string = 'GeneralPurpose'
+
+@description('Optional. Azure database for MySQL sku family')
+param skuFamily string = 'Gen5'
+
+@description('Optional. The source server resource id to restore from. It\'s required when "createMode" is "GeoRestore" or "Replica" or "PointInTimeRestore".')
+param sourceServerResourceId string = ''
+
+@description('Optional. Auto grow of storage.')
+param enableStorageAutogrow bool = true
+
+@description('Validate input parameter for storageAutogrow')
+var validStorageAutogrow = createMode == 'Replica' ? '' : (enableStorageAutogrow ? 'Enabled' : 'Disabled')
+
+@description('Optional. The storage size of the server.')
+param storageSizeGB int = 32
+
+@description('Optional. The version of the MySQL server.')
+@allowed(['5.6', '5.7', '8.0'])
+param version string = '8.0'
+
+@description('Optional. Array of role assignment objects that contain the "roleDefinitionIdOrName" and "principalId" to define RBAC role assignments on this resource.')
+param roleAssignments array = []
+
+// Should I share the same type defined in the modules/privateEndpoint?
```

Comment:
Any idea on how to share a type with modules/privateEndpoint?

Reply:
Not yet, but @jeskew , FYI +1 for request to loading types.